pilot/pkg/serviceregistry/kube/controller/multicluster.go

```diff
@@ -314,6 +319,40 @@ func (m *Multicluster) initializeCluster(cluster *multicluster.Cluster, kubeRegi
 	return nil
 }
 
+const (
+	externalIstiodLabel     = "externalIstiod"
+	istiodClusterAnnotation = "istiodClusterIDs"
```

Comment:
1. Do we need both annotations? Can we do a label selector with no `=true` to select things that have the label present? 
2. We should iterate on the name to something like `topology.istio.io/primaryCluster`
3. I'm not sure the mutating webhook is the place we want to store this important topology info in the long term, but it's fine for now.

Reply:
1. I've been wondering if it's really needed. It's really only used to optimize the selector for the query. It is, however, also an indicator that a cluster is istiodless, so maybe good to have in the future.
2. I actually started with `primaryCluster` but changed it because I thought it was confusing for external control planes, where the config cluster is the cluster ID that istiod uses, not a primary cluster.
3. The mutating webhook is certainly a convenient place to add the label, since it's a template in the remote cluster install. Putting on the namesapce, for example, is a little harder, since it would have to be done programmatically, or manually?

Reply:
Or ... maybe we don't need to identify an istiodless remote cluster. All we need is one label that identifies the clusters that should attempt remote leader election on any cluster (primary or remote, istiodless or not)?